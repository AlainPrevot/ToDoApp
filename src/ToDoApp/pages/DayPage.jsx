import { ToDoDate } from "../components/ToDoDate";

import { ToDoContentListAdd } from "../components/ToDoContentListAdd";

export const DayPage = () => {
  return (
    <>
      <ToDoDate />

      <ToDoContentListAdd />
    </>
  );
};
