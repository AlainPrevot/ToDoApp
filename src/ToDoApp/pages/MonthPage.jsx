import { ToDoContentListAdd } from "../components/ToDoContentListAdd";
import { ToDoDate } from "../components/ToDoDate";

export const MonthPage = () => {
  return (
    <>
      <ToDoDate />

      <ToDoContentListAdd />
    </>
  );
};
