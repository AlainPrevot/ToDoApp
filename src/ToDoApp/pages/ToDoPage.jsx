import { ToDoAdd, ToDoDate, ToDoList, ToDoNav } from "../components";
import { useToDo } from "../hooks";

import "../../sass/pages/ToDoPage.sass";

export const ToDoPage = () => {
  const {
    todos,
    // todosCount,
    // pendingTodosCount,
    handleDeleteTodo,
    handleToggleTodo,
    handleNewTodo,
  } = useToDo();

  return (
    <>
      <div className="ToDoPage">
        <ToDoNav />
        <div className="toDoPage__div--add">
          <ToDoDate />
          <ToDoAdd onNewTodo={handleNewTodo} />
        </div>
        <div className="toDoPage__Div--list">
          <ToDoList
            todos={todos}
            onDeleteTodo={handleDeleteTodo}
            onToggleTodo={handleToggleTodo}
          />
        </div>
      </div>
    </>
  );
};
