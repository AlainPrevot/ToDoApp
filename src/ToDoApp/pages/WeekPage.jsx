import { ToDoContentListAdd } from "../components/ToDoContentListAdd";
import { ToDoDate } from "../components/ToDoDate";

export const WeekPage = () => {
  return (
    <>
      <ToDoDate />
      <hr />

      <ToDoContentListAdd />
    </>
  );
};
