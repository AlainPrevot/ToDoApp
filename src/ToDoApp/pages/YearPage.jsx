import { ToDoContentListAdd } from "../components/ToDoContentListAdd";
import { ToDoDate } from "../components/ToDoDate";

export const YearPage = () => {
  return (
    <>
      <ToDoDate />

      <ToDoContentListAdd />
    </>
  );
};
