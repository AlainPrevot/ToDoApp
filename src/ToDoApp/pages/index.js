export * from './DayPage.jsx'
export * from './WeekPage.jsx'
export * from './MonthPage.jsx'
export * from './YearPage.jsx'