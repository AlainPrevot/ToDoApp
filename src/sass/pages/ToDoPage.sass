body
    background-color: #6D9AB3
    overflow: hidden
    height: 100vh
    display: flex
    align-items: center
    justify-content: center
    color: white

.ToDoPage
    background-color: #213140
    width: 620px
    height: 700px
    border-radius: 6px
    display: flex
    flex-direction: column
    align-items: center

    .toDoPage__div--add
        width: 90%

    .toDoPage__Div--list
        margin-top: 20px
        width: 90%

    .toDoPage__nav
        border: 1px solid red
        width: 100%
        height: 40px
