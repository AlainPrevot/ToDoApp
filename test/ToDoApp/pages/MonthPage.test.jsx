/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import { MonthPage } from "../../../src/ToDoApp/pages/MonthPage";

describe("Pruebas en <DayPage/>", () => {
  Object.defineProperty(window, "location", {
    value: {
      pathname: "/day",
    },
    writable: true,
  });

  test("Debe de hacer match con el Snapshot", () => {
    const { container } = render(<MonthPage />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrarse correctamente", () => {
    const { container } = render(<MonthPage />);

    expect(container.querySelector(".ToDoContentListAdd")).toBeTruthy();
    expect(container.querySelector(".toDoDate")).toBeTruthy();
  });
});
