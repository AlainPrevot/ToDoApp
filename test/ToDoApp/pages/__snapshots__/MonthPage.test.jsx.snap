// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DayPage/> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="toDoDate"
  >
    <h2>
      Thursday
    </h2>
    <h6>
      Dec
       
      14
      , 
      2023
    </h6>
  </div>
  <div
    class="ToDoContentListAdd"
  >
    <form>
      <input
        autocomplete="off"
        class=" toDoAdd__input"
        name="description"
        placeholder="Add your next task..."
        type="text"
        value=""
      />
      <button
        class="toDoAdd__button"
        type="submit"
      >
        Add
      </button>
    </form>
    <div
      class="toDoContentListAdd__div"
    >
      <ul
        class="toDoList"
      />
    </div>
  </div>
</div>
`;
